language/tools/move-cli/src/main.rs

```diff
@@ -467,29 +467,45 @@ fn explain_execution_effects(
     }
     for (addr, account) in &changeset.accounts {
         print!("  ");
+        if account.resources.is_empty() {
+            continue;
+        }
         println!(
             "Changed {:?} resource(s) under address {:?}:",
             account.resources.len(),
             addr
         );
+        let mut total_bytes_written = 0;
         for (struct_tag, write_opt) in &account.resources {
             print!("    ");
             match write_opt {
                 Some(blob) => {
+                    let bytes_to_write = blob.len() + struct_tag.access_vector().len();
+                    total_bytes_written += bytes_to_write;
                     if state
                         .get_resource_bytes(*addr, struct_tag.clone())?
                         .is_some()
                     {
                         // TODO: print resource diff
-                        println!("Changed type {}: {:?}", struct_tag, blob)
+                        println!(
+                            "Changed type {}: {:?} (wrote {:?} bytes)",
+                            struct_tag, blob, bytes_to_write
+                        )
                     } else {
                         // TODO: nicer printing
-                        println!("Added type {}: {:?}", struct_tag, blob)
+                        println!(
+                            "Added type {}: {:?} (wrote {:?} bytes)",
+                            struct_tag, blob, bytes_to_write
+                        )
                     }
                 }
                 None => println!("Deleted type {}", struct_tag),
```

Comment:
Just realized, does this need a `(wrote {:?} bytes)` too?

Reply:
Good question--it depends on how the storage backend works. In the Diem one, this will write sizeof(tag) bytes + a tombstone value (which I think is one byte)? I think at least accounting for the size of the tag makes sense here.

Reply:
Fixed.

Reply:
Thanks Sam! All others LGTM, feel free to land.